pkg/onchain/website/uploader.go

```diff
@@ -12,37 +12,26 @@ import (
 	"github.com/massalabs/thyra/pkg/wallet"
 )
 
-func PrepareForUpload(url string) (string, error) {
+func PrepareForUpload(url string, walletNickname string) (string, error) {
 	client := node.NewDefaultClient()
 
 	// TODO should use a nickname argument
-	wallet, err := wallet.FirstWallet()
+	wallet, err := wallet.GetWallet(walletNickname)
 	if err != nil {
 		return "", err
 	}
 
 	// Prepare address to webstorage.
-	scAddress, err := onchain.DeploySC(client, wallet, []byte(sc.WebsiteStorer))
+	scAddress, err := onchain.DeploySC(client, *wallet, []byte(sc.WebsiteStorer))
```

Comment:
Dereferencing the pointer makes the code look weird => this function should accept a pointer of wallet to avoir useless structure duplicate.